### PR TITLE
CAMEL-17586: Throw exception in case of duplicate attachment names

### PR DESCRIPTION
**Pull request by mash-sap**

### CODE REVIEW

File: components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java [hunk 1]
```diff
@@ -368,13 +368,18 @@ protected void extractAttachmentsFromMultipart(Multipart mp, Map<String, Attachm
...
-                        LOG.warn("Cannot extract duplicate file attachment: {}.", fileName);
+                        handleDuplicateAttachmentFilename(fileName);
                     }
                 }
             }
         }
     }
 
+    protected void handleDuplicateAttachmentFilename(String fileName) throws MessagingException {
+        LOG.error("Cannot extract duplicate file attachment: {}.", fileName);
```
Comment (davsclaus) [inline components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java:379]: Do not log an throw exception, only throw exception

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java [hunk 2]
```diff
@@ -368,13 +368,18 @@ protected void extractAttachmentsFromMultipart(Multipart mp, Map<String, Attachm
                         }
                         map.put(fileName, camelAttachment);
                     } else {
-                        LOG.warn("Cannot extract duplicate file attachment: {}.", fileName);
+                        handleDuplicateAttachmentFilename(fileName);
```
Comment (oscerd) [inline components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java:371]: I would avoid creating a method for one line exception throw

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a minor finding

**Comment by davsclaus** [comment]
I think its better to add an option you can configure to select what should happen

onDuplicateAttachment=warn|error

Then it can be warn as default which is what is today. And you can configure it to fail to have it throw the exception

File: components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java (hunk 2 above)
Comment (mash-sap) [reply components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java:371]: The idea was, that custom bindings could override the error handling, i.e. only log an error.

**Comment by mash-sap** [comment]
> 

Yes, this is probably a good idea. What would we need to do in order to get this option?

**Comment by davsclaus** [comment]
Adding new options is a bit more work, as you need to add it on the endpoint (and to make it global then on the component). Then you need to set that option on the binding as well, which is taken from the endpoint.

This gives the end users the freedom to set this globally on the component, or just quickly per endpoint.

The option can be a enum then you need to create an enum type, or it can also just be a plain string where you then check if its XX or YY 

And the option needs documentation, which you can do like its done in the other options.

**Comment by mash-sap** [comment]
> Contributor

Thanks for clarification. We currently discuss this in our team and we will update the pull request accordingly, if we continue with this approach.

**Comment by davsclaus** [comment]
Any update on this?

**Comment by davsclaus** [comment]
closing stale and out dated PR

**Pull request closed without merging**